### PR TITLE
add pre hook

### PR DESCRIPTION
**Pull request by mattapperson**
Use case:
Dynamically setting a value such as an API token that might be stored in local storage. This value might change over the use of the app, this allows injection of such a value without having to re-create every reducer and action in the app.

### CODE REVIEW

**Comment by lexich** [comment]
I understand your idea and it's ok. But 'pre' is a bad name for hook. It's short but misunderstandable. Maybe 'preprocessHook' or some other... And can you add tests and short description to readme. And tnx for your PR.

**Comment by mattapperson** [comment]
Will do

**Comment by lexich** [comment]
I think that the best way is options params can be function or object. If you need dynamic fuctionality your use function in other cases object

**Comment by mattapperson** [comment]
I like that idea. We could do the same with headers.

File: src/actionFn.js [hunk 1]
```diff
@@ -9,6 +11,12 @@ export default function actionFn(url, name, options, ACTIONS={}, fetch) {
     dispatch({ type: actionFetch });
     const _url = urlTransform(url, pathvars);
     const opts = { ...options, ...params };
+    if (opts.header) {
+      opts.header = isFunction(opts.header) ? opts.header() : opts.header;
+    }
+    if (opts.body) {
+      opts.body = isFunction(opts.body) ? opts.body() : opts.body;
+    }
```
Comment (lexich) [inline src/actionFn.js]: I think that the right way make options params as function.
```js
const opts = {
  ...isFunction(options) ? options() : options,
  ...params
}
```

**Comment by lexich** [comment]
closed by #7

**Pull request closed without merging**